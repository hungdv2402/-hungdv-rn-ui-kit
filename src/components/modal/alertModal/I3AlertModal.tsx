import React from 'react';
import { Pressable, StyleSheet, View, Text } from 'react-native';
import Modal from 'react-native-modal';
import SizeBox from '../../sizeBox';

interface I3AlertModalProps {
  title?: string;
  visible: boolean;
  subTitle?: string;
  negativeText?: string;
  positiveText?: string;
  onCancel(): void;
  onSubmit(): void;
}

const I3AlertModal = ({
  title,
  visible,
  subTitle,
  negativeText,
  positiveText,
  onCancel,
  onSubmit,
}: I3AlertModalProps) => {
  return (
    <Modal
      animationIn={'zoomIn'}
      animationOut={'zoomOut'}
      useNativeDriverForBackdrop
      hasBackdrop
      isVisible={visible}
    >
      <View style={styles.container}>
        <View style={styles.modal}>
          {title && <Text style={styles.title}>{title}</Text>}
          <SizeBox height={16} />
          {subTitle && <Text style={styles.content}>{subTitle}</Text>}
          <SizeBox height={40} />

          <View style={styles.buttonWrapper}>
            <Pressable onPress={onCancel} style={[styles.button, styles.cancelButton]}>
              <Text style={[styles.buttonText, { color: '#606163' }]}>{negativeText}</Text>
            </Pressable>

            <Pressable onPress={onSubmit} style={styles.button}>
              <Text style={styles.buttonText}>{positiveText}</Text>
            </Pressable>
          </View>
        </View>
      </View>
    </Modal>
  );
};

export default I3AlertModal;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  modal: {
    backgroundColor: '#fff',
    padding: 24,
    borderRadius: 16,
  },
  title: {
    fontSize: 16,
    fontWeight: '600',
    textAlign: 'center',
  },
  button: {
    height: 40,
    width: 130,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 20,
    backgroundColor: '#16AB89',
  },
  content: {
    fontSize: 14,
    textAlign: 'center',
  },
  buttonText: {
    fontWeight: '600',
    fontSize: 14,
    color: '#fff',
  },
  cancelButton: {
    backgroundColor: '#fff',
    borderWidth: 2,
    borderColor: '#F0F0F0',
  },
  buttonWrapper: {
    flexDirection: 'row',
    gap: 8,
  },
});
