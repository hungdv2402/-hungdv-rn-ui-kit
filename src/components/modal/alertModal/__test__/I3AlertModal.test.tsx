import renderer from 'react-test-renderer';
import I3AlertModal from '../I3AlertModal';
import React from 'react';

jest.mock('react-native-modal', () => 'Modal');

describe('renders I3AlertModal correctly', () => {
  it('should render I3AlertModal with basic properties', () => {
    const tree = renderer
      .create(
        <I3AlertModal
          visible={false}
          onCancel={jest.fn()}
          onSubmit={jest.fn()}
          title={'title'}
          subTitle={'subTitle'}
          negativeText={'negativeText'}
          positiveText={'positiveText'}
        />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});