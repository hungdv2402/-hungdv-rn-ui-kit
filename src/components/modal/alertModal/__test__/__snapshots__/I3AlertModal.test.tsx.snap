// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders I3AlertModal correctly should render I3AlertModal with basic properties 1`] = `
<Modal
  animationIn="zoomIn"
  animationOut="zoomOut"
  hasBackdrop={true}
  isVisible={false}
  useNativeDriverForBackdrop={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 16,
          "padding": 24,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        title
      </Text>
      <View
        style={
          {
            "height": 16,
            "width": undefined,
          }
        }
      />
      <Text
        style={
          {
            "fontSize": 14,
            "textAlign": "center",
          }
        }
      >
        subTitle
      </Text>
      <View
        style={
          {
            "height": 40,
            "width": undefined,
          }
        }
      />
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#16AB89",
                "borderRadius": 20,
                "height": 40,
                "justifyContent": "center",
                "width": 130,
              },
              {
                "backgroundColor": "#fff",
                "borderColor": "#F0F0F0",
                "borderWidth": 2,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#fff",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
                {
                  "color": "#606163",
                },
              ]
            }
          >
            negativeText
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#16AB89",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "width": 130,
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 14,
                "fontWeight": "600",
              }
            }
          >
            positiveText
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
